ui/src/app/views/profile-page/component.ts

```diff
@@ -55,14 +58,24 @@ export class ProfilePageComponent implements OnInit {
     this.profileService.getMe().subscribe(
       (profile: Profile) => {
         this.profile = profile;
+        this.workingProfile = <Profile> deepCopy(profile);
         this.profileLoaded = true;
         this.reloadSpinner();
       });
   }
 
   submitChanges(): void {
-    this.profileService.updateProfile(this.profile).subscribe(
+    if (this.workingProfile.institutionalAffiliations !== undefined) {
+      let currentSpot = 0;
+      this.workingProfile.institutionalAffiliations.forEach(
```

Comment:
Two things:

First a nit, you've got the indentation for the function signature and body reversed.  Put the body at a higher indentation level.

Second, `forEach` (and basically almost all other js array iteration methods) provides the index of the element as a second argument to the callback.  So in other words, this whole block can be boiled down to:

```
affiliations.forEach((affiliation, index) => affiliation.orderIndex = index);
```

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach